src/routes/friend.js

```diff
@@ -1,88 +1,164 @@
 const router = require('express').Router();
 const Student = require('../models/student.model');
 const StudentRepository = require('../repository/studentRepository')
+const FriendRequest = require('../models/friendRequest.model');
+const createFriendRequest = require('../factory/friendRequestFactory');
+const FriendRequestRepository = require('../repository/friendRequestRepository');
 
-// fetch the list of friends by email
+/**
+ * @author: Jasmin Guay
+ * Endpoint to send a friend request.
+ * @param {string} senderEmail : email of the sender
+ * @param {string} receiverEmail : email of the students to be added
+ * @return {string} _id  : id of the friend request created.
+ */
+router.route('/add').post(async (req, res) => {
+  const senderEmail = req.body.senderEmail.toString();
+  const receiverEmail = req.body.receiverEmail.toString();
+
+  if (!await StudentRepository.isInFriendList(senderEmail, receiverEmail)) {
+    const newFriendRequest = createFriendRequest({senderEmail, receiverEmail});
+    FriendRequestRepository.save({senderEmail, receiverEmail})
+        .then((res) => res.json(result._id).status(200))
```

Comment:
also result is undefined
this should be `.then((result) => res.json(result._id).status(200))`